libs/storage/Tsavorite/cs/src/core/Allocator/TsavoriteLogAllocatorImpl.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Tsavorite.core
+{
+    // This is unused; just allows things to build. TsavoriteLog does not do key comparisons or value operations; it is just a memory allocator.
+    using TsavoriteLogStoreFunctions = StoreFunctions<SpanByteComparer, DefaultRecordDisposer>;
+
+    /// <summary>Simple log allocator used by TsavoriteLog</summary>
+    public sealed unsafe class TsavoriteLogAllocatorImpl : AllocatorBase<TsavoriteLogStoreFunctions, TsavoriteLogAllocator>
+    {
+        private readonly OverflowPool<PageUnit<Empty>> freePagePool;
+
+        /// <summary>Constructor</summary>
+#pragma warning disable IDE0290 // Use primary constructor
+        public TsavoriteLogAllocatorImpl(AllocatorSettings settings)
+            : base(settings.LogSettings, new TsavoriteLogStoreFunctions(), @this => new TsavoriteLogAllocator(@this), evictCallback: null, settings.epoch, settings.flushCallback, settings.logger)
+        {
+            freePagePool = new OverflowPool<PageUnit<Empty>>(4, p => { });
```

Comment:
Ditto, do unmanaged pointers end up in this pool?

Reply:
Yes, and this is cleared and they are freed by AllocatorBase during Dispose()